packages/strapi-connector-bookshelf/lib/mount-models.js

```diff
@@ -665,6 +666,13 @@ module.exports = ({ models, target }, ctx) => {
       if (err instanceof TypeError || err instanceof ReferenceError) {
         strapi.stopWithError(err, `Impossible to register the '${model}' model.`);
       }
+
+      if (['ER_TOO_LONG_IDENT'].includes(err.code)) {
+        strapi.stopWithError(
+          err,
+          'A table name is too long. If it is the name of a join table automatically generated by Strapi, you can customise it by adding `collectionName: "customName"` in the corresponding attribute model (on the dominant side if it\'s a many-to-many relation).'
```

Comment:
... In the corresponding model's attribute. 
When this happens on a manyToMany relation, make sure to set this parameter on the dominant side of the relation (e.g: where `dominant: true` is set)